odoo/addons/base/models/res_users.py

```diff
@@ -473,7 +473,7 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
     def create(self, vals_list):
         users = super(Users, self).create(vals_list)
         for user in users:
-            user.partner_id.write({'company_id': user.company_id.id, 'active': user.active})
+            user.partner_id.write({'active': user.active})
```

Comment:
diff --git odoo/addons/base/models/res_users.py odoo/addons/base/models/res_users.py
index f746fb1f678..a826f82d694 100644
--- odoo/addons/base/models/res_users.py
+++ odoo/addons/base/models/res_users.py
@@ -473,7 +473,10 @@ class Users(models.Model):
     def create(self, vals_list):
         users = super(Users, self).create(vals_list)
         for user in users:
-            user.partner_id.write({'company_id': user.company_id.id, 'active': user.active})
+            # if partner is global we keep it that way
+            if user.partner_id.company_id:
+                user.partner_id.company_id = user.company_id
+            user.partner_id.active = user.active
         return users

Reply:
Oh right! We can pass some values when create method is called, so it's possible that the partner has already a company_id.
I have updated the code and I have added a new test to check when a partner with a different company is given at user creation.